apps/condo/pages/_app.tsx

```diff
@@ -77,44 +79,55 @@ const MenuItems: React.FC = () => {
     const disabled = !link || (hasSubscriptionFeature && isExpired)
     const { isCollapsed } = useLayoutContext()
     const { wrapElementIntoNoOrganizationToolTip } = useNoOrganizationToolTip()
+    const role = get(link, 'role', {})
 
-    const menuItemsData: IMenuItemData[] = [{
-        path: 'reports',
-        icon: BarChartIconNew,
-        label: 'menu.Analytics',
-    }, {
-        path: 'ticket',
-        icon: BarTicketIcon,
-        label: 'menu.ControlRoom',
-    }, {
-        path: 'property',
-        icon: BarPropertyIcon,
-        label: 'menu.Property',
-    }, {
-        path: 'contact',
-        icon: BarUserIcon,
-        label: 'menu.Contacts',
-    }, {
-        path: 'employee',
-        icon: BarEmployeeIcon,
-        label: 'menu.Employees',
-    }, {
-        path: 'billing',
-        icon: BarBillingIcon,
-        label: 'menu.Billing',
-    }, {
-        path: 'payments',
-        icon: BarPaymentsIcon,
-        label: 'menu.Payments',
-    }, {
-        path: 'meter',
-        icon: BarMeterIcon,
-        label: 'menu.Meters',
-    }, {
-        path: 'miniapps',
-        icon: BarMiniAppsIcon,
-        label: 'menu.MiniApps',
-    }]
+    const menuItemsData: IMenuItemData[] = useMemo(() => {
+        const itemsConfigs = [{
+            path: 'reports',
+            icon: BarChartIconNew,
+            label: 'menu.Analytics',
+        }, {
+            path: 'ticket',
+            icon: BarTicketIcon,
+            label: 'menu.ControlRoom',
+        }, {
+            path: 'property',
+            icon: BarPropertyIcon,
+            label: 'menu.Property',
+        }, {
+            path: 'contact',
+            icon: BarUserIcon,
+            label: 'menu.Contacts',
+        }, {
+            path: 'employee',
+            icon: BarEmployeeIcon,
+            label: 'menu.Employees',
+        }, {
+            path: 'billing',
+            icon: BarBillingIcon,
+            label: 'menu.Billing',
+            access: () => get(role, 'canReadBillingReceipts', false ),
+        }, {
+            path: 'payments',
+            icon: BarPaymentsIcon,
+            label: 'menu.Payments',
+            access: () => get(role, 'canReadPayments', false ),
+        }, {
+            path: 'meter',
+            icon: BarMeterIcon,
+            label: 'menu.Meters',
+        }, {
+            path: 'miniapps',
+            icon: BarMiniAppsIcon,
+            label: 'menu.MiniApps',
+        }]
+        const result = []
+        itemsConfigs.filter((item) => {
+            const itemAccesses = get(item, 'access', () => true)
+            return itemAccesses() && result.push(item)
+        })
+        return result
```

Comment:
You are already filtering menu items. It's not needed to build the same array again.
```suggestion

        return itemsConfigs.filter((item) => get(item, 'access', () => true)())
```